platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# implementation of new platform api
+#############################################################################
+
+try:
+    from sonic_platform_base.component_base import ComponentBase
+    from glob import glob
+    import subprocess
+    import io
+    import re
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+#components definitions
+COMPONENT_BIOS = "BIOS"
+COMPONENT_CPLD = "CPLD"
+
+BIOS_QUERY_VERSION_COMMAND = 'dmidecode -t 11'
+CPLD_VERSION_FILE_PATTERN = '/var/run/hw-management/system/cpld[0-9]_version'
+CPLD_VERSION_MAX_LENGTH = 4
+
+class Component(ComponentBase):
+    def get_name(self):
+        """
+        Retrieves the name of the component
+
+        Returns:
+            A string containing the name of the component
+        """
+        return self.name
+
+
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        result = ''
+        try:
+            with io.open(filename, 'r') as fileobj:
+                result = fileobj.read(len)
+            return result
+        except Exception as e:
+            logger.log_info("Fail to read file {} due to {}".format(filename, repr(e)))
+            return '0'
+
+
+    def _get_command_result(self, cmdline):
+        try:
+            proc = subprocess.Popen(cmdline, stdout=subprocess.PIPE, shell=True, stderr=subprocess.STDOUT)
+            stdout = proc.communicate()[0]
+            proc.wait()
+            result = stdout.rstrip('\n')
+
+        except OSError, e:
+            result = ''
```

Comment:
@stephenxs how do you suppose to handle errors here? IMHO, this is critical so the exception should be handled in a proper way. Also the same comment as above: consider moving to a separate helper class/namespace because this has nothing to do with the components.

Reply:
Having reflected it, I think upper layer of platform api is platform-independent and not supposed to know and handle platform-dependent details, like variant kinds of exceptions because such behavior seems to violate the pricinple of software layering and modularization.
But what you said is also reasonable, upper layer should be notified in case of this kind of errors. To achieve that, we have some options:
1. platform api returns a certain value indicating an error occurs, for example an empty string or None. upper layer test the return value and handle the error case.
2. platform api catch the platform-specific exceptions and raise a runtime error that will be handled by upper layer.
I prefer the second option since it's obvious an error here.
Meanwhile, I also suggest having platform api clearly defined regarding error return and handling.